### ISSUE TITLE
maintainers, future of cookieconsent

### ISSUE THREAD

**Issue by DanielRuf** [insites/cookieconsent#409]
Hi there, I would be interested in joining as maintainer as there are many open issues and PRs or create a fork to bring this project forward.

cc @Dolondro

**Comment by Ruud68** [comment]
Been following this project for some time now, been using the code myself. I see a lot of people forking, creating PR's for issues found and suggest (functional) improvements. The problem is not that these PR's are not pulled by the maintainers and that when you want to use this project you have to create your own fork and cherry pick from the many PR's. The problem is that there is no communication from the maintainers whatsoever (other then the fishing mail) leaving every body 'in the dark' and clueless as to what the status of this project is...

So I am very curious as to what is going on with this project. Why the maintainer(s) choose not to communicate. Is this project abandoned, are you to busy? If so, that's okay, just tell us....

@DanielRuf although I really would like you (or somebody else) to join the list of maintainers of this project and harvest the energy from all community contributors, I fully understand that due to lack of response you create and maintain a fork. If you decide to fork and maintain that fork and are open to PR's, I think you can count a lot of people in and your fork will really lift off very quick!

**Comment by DanielRuf** [comment]
I totally agree with your points @Ruud68

Definitely, we'll try to incorporate changes and PRs of different forks and bring the changes together to integrate the solutions and improvements and create new releases to better suit the needs and usage compared to closed source solutions.

**Comment by alexmorleyfinch** [comment]
I am totally up for some community driven maintenance, we need to discuss this internally before we can do anything though. We will be discussing this soon.

**Comment by DanielRuf** [comment]
> I am totally up for some community driven maintenance, we need to discuss this internally before we can do anything though. We will be discussing this soon.

Sounds good. Please keep us updated.

**Comment by alexmorleyfinch** [comment]
We voted against community maintenance for a few reasons:

1. CookieConsent (although popular) does not legally cover the requirements of [GDPR](https://en.wikipedia.org/wiki/General_Data_Protection_Regulation), therefore assigning (external) maintainers at this late stage seems pointless, as it is no longer compliant. It no longer offers any protection.

CookieConsent was originally designed to handle Europe's [ePrivacy Directive](https://en.wikipedia.org/wiki/Privacy_and_Electronic_Communications_Directive_2002). This directive is no longer in effect and has been superseded by [GDPR](https://en.wikipedia.org/wiki/General_Data_Protection_Regulation).

(As an aside, parts of GDPR may also be superseded again within a couple of years due to the [ePrivacy Regulation](https://en.wikipedia.org/wiki/EPrivacy_Regulation_(European_Union)) currently being drafted, which removes the need for banners completely for most people.

2. We will soon (within a month) be releasing a new plugin that satisfies the lawful requirements of GDPR, which will function similar to our existing CookieConsent plugin. This will officially deprecate CookieConsent (however we will still allow people to use it if they wish).

Because CookieConsent is no longer compliant, we will eventually aim to encourage people to move over to a solution that is compliant with the law. Seeing as we haven't completed our version of a solution yet, we've been hesitant to shout about it.

We do however understand that it will still take a while for the world to catch up and people to move to new solutions, so we are still welcoming contributions to CookieConsent.

A maintainer has been assigned to work through the backlog of PRs and issues to bring the repo up to standard.

Finally, an apology (to the community) for being quiet for so long. We have been very busy for the last year, but this is no excuse to ignore our responsibilities.

**Comment by xolan** [comment]
Any news on a GDPR-catered version? :)

**Comment by Dolondro** [comment]
The lead on it is currently off after having his wisdom teeth pulled, I'll update you once he's back in 👍

**Comment by HuubSmit** [comment]
Any news?

**Comment by Dolondro** [comment]
The latest news seems to be that it's mostly complete but we need to create the tool for easy configuration which is blocked by someone who has been sidelined onto a more business critical project for a few weeks :(

I'll put a note in my calendar and try to keep this updated

**Comment by flivni** [comment]
Any update on this?

**Comment by alexmorleyfinch** [comment]
We're still working on a basic version (along side working on other projects). V1 will be out soon for comments and criticism, but I cannot provide a date unfortunately

**Comment by niccoX** [comment]
> Because CookieConsent is no longer compliant,

Why is it not compliant, it does what the other banners do, informs you and lets you accept the settings?

**Comment by paire** [comment]
> 
> 
> > Because CookieConsent is no longer compliant,
> 
> Why is it not compliant, it does what the other banners do, informs you and lets you accept the settings?

GDPR has more stringent requirements than the EUCookieLaw. 

- Passive consent is no longer acceptable (the visitor has to opt-in, which the current version does allow you to do).

- The visitor has to be able to revoke consent (which the current version does not provide).

- The consent has to be renewed regularly (once a year is recommended). 

- Consent has to be documented/recorded (which doesn't happen currently).

- You have to be able to scrub any trace of the visitor when requested (good luck with that because you probably have no idea what cookie was set for a given user if they haven't given you their email address).

- You also have be be clear about how the cookies are being used (which is just wording on the consent form, no big deal). A number of companies are offering varying levels of opt-in (agree to marketing & analytics, agree to statistics, agree to preferences, etc. which the current version doesn't do); however, this is extra and is not required; you just have to be transparent about how their data will be used.

- The consent has to be given before any processing happens (within reason); if the cookie banner is still up and the visitor continues to browse your website, you can't use any cookies other than those that are strictly necessary (I don't know if the current version does this or not. I don't think so, but I think it depends on how it's implemented on your site).

**Comment by niccoX** [comment]
> > > Because CookieConsent is no longer compliant,
> > 
> > 
> > Why is it not compliant, it does what the other banners do, informs you and lets you accept the settings?
> 
> GDPR has more stringent requirements than the EUCookieLaw.
> 
> * Passive consent is no longer acceptable (the visitor has to opt-in, which the current version does allow you to do).

_I'm not sure that current GDPR law reflects this, I think you can still install cookies before the visitor accepts or rejects cookies. Sites I have tested on all install cookies the second you visit them, their consent is just for additional cookies to be installed._

> * The visitor has to be able to revoke consent (which the current version does not provide).
_But this can be explained in the policy, how the user can delete his cookies within the browser._

> * The consent has to be renewed regularly (once a year is recommended).

_Why if you haven't changed it?_

> * Consent has to be documented/recorded (which sounds like it happens via the API).

_I don't know, you mean the visitors choice has to be recorded?_

> * You have to be able to scrub any trace of the visitor when requested (good luck with that because you probably have no idea what cookie was set for a given user if they haven't given you their email address).

_?_

> * You also have be be clear about how the cookies are being used (which is just wording on the consent form, no big deal). A number of companies are offering varying levels of opt-in (agree to marketing & analytics, agree to statistics, agree to preferences, etc. which the current version doesn't do); however, this is extra and is not required; you just have to be transparent about how their data will be used.

> * The consent has to be given before any processing happens (within reason); if the cookie banner is still up and the visitor continues to browse your website, you can't use any cookies other than those that are strictly necessary (I don't know if the current version does this or not. I don't think so, but I think it depends on how it's implemented on your site).

_You can implement scroll or click which will equal that the visitor has accepted the policy._

**Comment by DanielRuf** [comment]
> * Passive consent is no longer acceptable (the visitor has to opt-in, which the current version does allow you to do).

Covered by cookieconsent. As long as you do not make any choice, it does nothing.

> * The visitor has to be able to revoke consent (which the current version does not provide).

Huh? We've done this in many projects here. On `dismiss` or `false` we hide it. And we can always change it.

> * The consent has to be renewed regularly (once a year is recommended).

Not sure what you mean with this.

> * Consent has to be documented/recorded (which sounds like it happens via the API).

Possibly with a simple AJAX request on acceptance / rejection (depends on ypour backend).

> * You have to be able to scrub any trace of the visitor when requested (good luck with that because you probably have no idea what cookie was set for a given user if they haven't given you their email address).

That's not what a cookie banner does and should do. This is dependent of the used CMS.

> * You also have be be clear about how the cookies are being used (which is just wording on the consent form, no big deal). A number of companies are offering varying levels of opt-in (agree to marketing & analytics, agree to statistics, agree to preferences, etc. which the current version doesn't do); however, this is extra and is not required; you just have to be transparent about how their data will be used.

We display an overlay and set a second cookie for the level.

> * The consent has to be given before any processing happens (within reason); if the cookie banner is still up and the visitor continues to browse your website, you can't use any cookies other than those that are strictly necessary (I don't know if the current version does this or not. I don't think so, but I think it depends on how it's implemented on your site).

We have checks for this and DNT in our scripts and backend code.

**Comment by DanielRuf** [comment]
> I'm not sure that current GDPR law reflects this, I think you can still install cookies before the visitor accepts or rejects cookies. Sites I have tested on all install cookies the second you visit them, their consent is just for additional cookies to be installed.

There is a difference between tracking cookies and required cookies.

**Comment by paire** [comment]
> > * The visitor has to be able to revoke consent (which the current version does not provide).
> 
> Huh? We've done this in many projects here. On `dismiss` or `false` we hide it. And we can always change it.

Sorry, for publishing bad info. Good to know. I wasn't aware that the visitor themselves could revoke their consent after it's been granted, instead of requiring you to do it for them. (Which is my understanding of how this is supposed to work. If that understanding is wrong, I'm happy to be corrected.)

> 
> > * The consent has to be renewed regularly (once a year is recommended).
> 
> Not sure what you mean with this.

Again, sorry. I think I was mixing e-Privacy Directive recommendations with GDPR requirements. The GDPR regulations (from what I understand) state that old consents may need to be renewed (since the old consent was pre-GDPR and therefore you may have had users opt-in for something in a way that's not GDPR complaint - such as not keeping track of their consent). If your prior implementation took care of all the GDPR requirements then you don't have to renew their consent. 


> > * The consent has to be given before any processing happens (within reason); if the cookie banner is still up and the visitor continues to browse your website, you can't use any cookies other than those that are strictly necessary (I don't know if the current version does this or not. I don't think so, but I think it depends on how it's implemented on your site).
> 
> We have checks for this and DNT in our scripts and backend code.

Cool, thanks for the info.

(And thanks for the quick response.)

**Comment by paire** [comment]
> 
> 
> > I'm not sure that current GDPR law reflects this, I think you can still install cookies before the visitor accepts or rejects cookies. Sites I have tested on all install cookies the second you visit them, their consent is just for additional cookies to be installed.
> 
> There is a difference between tracking cookies and required cookies.

Agreed, no consent is needed for necessary cookies. They're fine as soon as the visitor hits the site.

**Comment by DanielRuf** [comment]
> Sorry, for publishing bad info. Good to know. I wasn't aware that the visitor themselves could revoke their consent after it's been granted, instead of requiring you to do it for them. (Which is my understanding of how this is supposed to work. If that understanding is wrong, I'm happy to be corrected.)

There is an optin / optout feature already implemented which sets `undefined`, `true` or `false` depending on your choice.

**Comment by relicmelex** [comment]
Hello to everyone following this post. I'll quickly introduce myself. I'm Eric, I'm a JavaScript nerd and I'm here to help! I'm your new point of contact maintainer. Please feel free to @ me anytime here and I'll do my best to answer you in a prompt and detailed manner.

First thank you for your continued support of this project, all the conversation and suggestions are amazing. Now that I'm here we will be resolving a lot of the current bugs, slating features for v3, modernizing in v4 and new features and integrations in v4.X

You'll notice me cruising through the current backlog of issues. I'm going to try and have them all responded too by the end of the month, a bunch of bugs resolved, and probably some features as well. and ready for the next release. This isn't a promise on the 3.1.2 release date or anything, I'm just saying stay tuned.

Oh yes, I'm locking this issue because it's gotten way off topic and I'm handling individual issues now regularly. I encourage you to reach out if you have questions or concerns or with your pull requests. However I ask you to create issues for feature requests or enhancements so there can be discussion around implementation before execution so making PRs are in the intended direction of CookieConsent.
